main.go

```diff
@@ -106,6 +106,68 @@ func signTxCmd() *ffcli.Command {
 	}
 }
 
+func batchSignTxCmd() *ffcli.Command {
+	fs := flag.NewFlagSet("sign-batch", flag.ContinueOnError)
+	keyringDir := fs.String("keyring-dir", "", "Keyring directory (mandatory with -keyring-backend=file)")
+	keyringBackend := fs.String("keyring-backend", "", "Keyring backend, which can be one of 'keychain' (macos), 'pass', 'kwallet' (linux), or 'file'")
+	signer := fs.String("from", "", "Signer key name")
+	chainID := fs.String("chain-id", "", "Chain identifier")
+	account := fs.String("account", "", "Account number")
+	sequence := fs.String("sequence", "", "Sequence number")
+	sigOnly := fs.Bool("signature-only", false, "Outputs only the signature data")
+	return &ffcli.Command{
+		Name:       "sign-batch",
+		ShortUsage: "json-signer sign-batch -from=<key> -keyring-backend=<keychain|pass|kwallet|file> -chain-id=<chainID> -sequence=<sequence> -account=<account-number> <tx.json>",
+		ShortHelp:  "Sign batch transaction",
+		FlagSet:    fs,
+		Exec: func(ctx context.Context, args []string) error {
+			if err := fs.Parse(args); err != nil {
+				return err
+			}
+			if fs.NArg() != 1 || *keyringBackend == "" || *signer == "" ||
+				*sequence == "" || *account == "" || *chainID == "" {
+				return flag.ErrHelp
+			}
+			if *keyringBackend == "file" && *keyringDir == "" {
+				return fmt.Errorf("-keyring-backend=file requires -keyring-dir flag")
+			}
+			kr, err := keyring.New(keyring.BackendType(*keyringBackend), *keyringDir, nil)
+			if err != nil {
+				return err
+			}
+
+			signedTxs, _, err := batchSignTx(kr, fs.Arg(0), *signer, *chainID, *account, *sequence)
```

Comment:
What about reading all the txs in this function, and call the existing `signTx` (so no need for `batchSignTx`) ? I'm not sure it's possible, but if it is, it keeps the code simpler.

Reply:
I ended up keeping a helper function just for ease of testing, but simplified it dramatically.